### PR TITLE
One-Click Update & Uninstall + Grouped Bar Chart + Many fixes

### PR DESCRIPTION
**Pull request by ipitio**
Aha! I knew I could download more speed.

How it started:

* arevindh/pihole-speedtest#125

How it's going:
* arevindh/pihole-speedtest#130
* arevindh/pi-hole#51
* This PR

Features:
* New bar chart
  * Switch per-browser and
  * Change the default for new browsers right from settings
* Mod the Mod
  * Update and/or [Un,Re]Install directly from settings
  * Failsafe
  * Offline uninstall
* Speedtest
  * Run a speedtest manually with a button in settings

Fixes:

* Chart
  * Set fixed height
  * Revamped colors
  * Re-resensitized tooltip
  * Put axes in their rightful places
  * Fixed mobile display issues
  * Removed legend
  * Hide date when only today
* Settings
  * Improved layout
  * Changed Save to Confirm and colored it red when history is to be cleared
  * Removed script selector as speedtest-cli is no longer supported
  * Properly handled custom server input, or lack thereof
* Speedtest
  * Properly handled failure
  * Read results once

Before:

image

After:

image

image

image

### CODE REVIEW

**Comment by arevindh** [comment]
Thanks for the PR. Will check the tomorrow, out of station now.

**Comment by arevindh** [comment]
@ipitio is it ready to test ?

**Comment by arevindh** [comment]
@ipitio 
Tested it and it is , working 👍  
Incase you are interested, we can chat here https://discord.gg/MmXREXEB

**Comment by ipitio** [comment]
@town3r there u go

**Comment by town3r** [comment]
Would it be hard to make it so the day of month/date doesn't show up on the horizontal graph axis if the timerange that's selected is last 24 hours?

**Comment by ipitio** [comment]
Could you see if that does it, my last 24 hours includes yesterday and I don't wanna flush

**Comment by town3r** [comment]
> Could you see if that does it, my last 24 hours includes yesterday and I don't wanna flush

🙈 Unfortunately, I don't know how to do that. 
I was just going to wait for https://github.com/arevindh/pihole-speedtest/issues/128 and this to go live and utilize the One-click Update option to get things all installed.

**Comment by ipitio** [comment]
> > Could you see if that does it, my last 24 hours includes yesterday and I don't wanna flush
> 
> 🙈 Unfortunately, I don't know how to do that. I was just going to wait for [arevindh/pihole-speedtest#128](https://github.com/arevindh/pihole-speedtest/issues/128) and this to go live and utilize the One-click Update option to get things all installed.

```
sudo rm -rf /var/www/html/org_admin
sudo mv /var/www/html/admin /var/www/html/org_admin
sudo git clone -b patch-1 https://github.com/ipitio/AdminLTE/ /var/www/html/admin
```

**Comment by town3r** [comment]
Thank you so much!
Looks like the date/day is still present.
IMG_A8E5FBE54E8D-1

**Comment by town3r** [comment]
Thank you for making the time axis tilted on mobile.

**Comment by ipitio** [comment]
@town3r try now

**Comment by town3r** [comment]
Looks good!
image

**Comment by ipitio** [comment]
Wow those tests are both random and so frequent lol

**Comment by town3r** [comment]
Oh to force some data onto the graph i ran the following every minute a bunch:

Test Pihole speedtest from commandline
`sudo "/var/www/html/admin/scripts/pi-hole/speedtest/speedtest-official.sh" `

**Comment by town3r** [comment]
The date is back ☹️ 

image

**Comment by ipitio** [comment]
> The date is back ☹️ 
> 
> image

Because you got the 19th over there on the left. 

I thought if "today" isn't the only date to be shown, even if less than 24 hours apart, it should still show the date. 

What do you think?

**Comment by town3r** [comment]
Eh I personally know the date and want it to match the other graphs if possible.

**Comment by ipitio** [comment]
> match the other graphs.

Whoops I had tunnel vision

**Comment by ipitio** [comment]
That should do it

**Comment by town3r** [comment]
Looks good

**Comment by ipitio** [comment]
@town3r You can test the update and uninstall buttons now! 

Just click on one and/or both of them and wait for the magic to happen. Or you can run `sudo tmux attach-session -t pimod` to see the progress. The update won't actually work until sid's next release, so you can think of it as a reinstall for now. 

And should you wonder what to do after uninstalling the mod, just run
```
sudo tmux new-session -s piup 'curl -sSL https://github.com/ipitio/pihole-speedtest/raw/ipitio/mod.sh | tac | tac | bash -s -- up'
```
and everything should be back as it was.

**Comment by town3r** [comment]
> sudo rm -rf /var/www/html/org_admin
> sudo mv /var/www/html/admin /var/www/html/org_admin
> sudo git clone -b patch-1 https://github.com/ipitio/AdminLTE/ /var/www/html/admin

Would/should I still use the above to upgrade to the latest version to see the Upgrade & Uninstall buttons?

**Comment by ipitio** [comment]
That curl command in my last comment is the only command you need now.

After accidentally deleting my FTL database, and some other fun adventures, I think my update script is finally worthy of non-beta status.

**Comment by ipitio** [comment]
Changed the layout of settings a bit.

**Comment by town3r** [comment]
Hmm something seems to have broken last night.

Been trying the fancy (Re)install Latest toggle under settings when I see you've pushed a new update.

Upon checking the graph this morning tho. It seems the test didn't run hourly like it normally does. Just kinda randomly.

image

**Comment by town3r** [comment]
Might be related to the Chart Display Range?

image

Eh no, that's also showing that the tests stopped at 08:00.

**Comment by ipitio** [comment]
@town3r can you install latest and try again

**Comment by town3r** [comment]
Hmm hourly tests might be broken again.

Are you able to add a `Every 10 Minutes` option for Testing Schedule?

**Comment by ipitio** [comment]
> Hmm hourly tests might be broken again.

Working now!

> Are you able to add a `Every 10 Minutes` option for Testing Schedule?

That's out of the scope of this PR, only doing bugfixes now.

File: scripts/pi-hole/speedtest/speedtest.sh [hunk 1]
```diff
@@ -1,326 +1,82 @@
...
-    server_dist=$(echo $server | sed 's/.*\[//g' | sed 's/\].*//g')
-    server=$(echo $server | sed 's/ \[.*//g')
-
-    download=$(str_extract "Download: ")
-    upload=$(str_extract "Upload: ")
-    if [[ $share == true ]]; then
-        share_url=$(str_extract "Share results: ")
+tryagain(){
+    if grep -q official <<< "$(/usr/bin/speedtest --version)"; then
+        apt-get install -y speedtest- speedtest-cli
```
Comment (arevindh) [inline scripts/pi-hole/speedtest/speedtest.sh:65]: This might not be possible in all cases (armv7 and aarch64 don't have packages )

Comment (arevindh) [reply scripts/pi-hole/speedtest/speedtest.sh:65]: `speedtest-cli` should not be installed since its the python version

**Pull request merged**